app/utils/lnurl.ts

```diff
@@ -0,0 +1,57 @@
+import { bech32 } from "bech32"
+
+import { satToMsat } from "./amount"
+
+import { SUPPORTED_PROTOCOLS } from "../constants/lnurl"
+
+import type { LNUrlPay } from "../types/lnurl"
+
+export const isLnUrl = (invoice: string): boolean =>
+  invoice.toLowerCase().startsWith("lnurl")
+
+export const isProtocolSupported = (tag: string): boolean => !!SUPPORTED_PROTOCOLS[tag]
+
+export const isAmountValid = (
+  amount: number,
+  minSendable: number,
+  maxSendable: number,
+): boolean => minSendable <= amount && amount <= maxSendable
+
+export const parseUrl = async (invoice: string): Promise<LNUrlPay> => {
+  const { words } = bech32.decode(invoice, 2000)
```

Comment:
Why was the number 2000 chosen? Is there a standard? I notice Blue Wallet uses 10000.

Reply:
This is not standard. The recommendation is to not exceed 1.000 characters because of error correction, and most of the lnurl-pay codes will not exceed 2.000 characters, so it seems reasonable to use this value.